### ISSUE TITLE
Webcam display extension

### ISSUE THREAD

**Issue by becky-gilbert** [lookit/lookit-jspsych#39]
## TL;DR

We need to decide how to add the webcam feed into various plugins, including whether/how to make it available as an option for researchers (a la the `showWebcam` parameter in various EFP frames). 

## Summary

Our `record` package does not currently support displaying the webcam feed.

In EFP, there are certain frames that always show the webcam feed to the participant:

- `video-consent`
- `video-config`
- `video-config-quality`
- `webcam-display`
- `observation` (researchers can initially show/hide the webcam feed, but it is always on the page and participants can click a button to show it)

There are also some frames that allow the researcher to _optionally_ show the webcam feed to the participant:

- `video-assent`
- `instructions`

For some of the frames in the list above, it will make sense for us to create a custom plugin that does the same thing as the EFP frame (e.g. `video-consent`, `video-config`). In these cases, it will be easy to incorporate the webcam display and add any necessary parameters to the plugin.

However, in other cases, it would be better if we could avoid re-creating existing jsPsych plugins that could be used for most of the same functionality, and just provide an extension that allows researchers to add the webcam feed to a non-CHS plugin. One example of this is the `instructions` EFP frame, which could easily reproduce with the existing jsPsych `instructions` plugin. 

## Questions

1. Is it ok to have these two different methods that result in the webcam feed appearing on the page (default/parameterized in plugin, and using an extension)?
2. What is the best/easiest way to allow researchers to insert a webcam into a non-CHS plugin and control its size/location? (BECKY TO ADD MORE DETAILS ABOUT OPTIONS)

## To do

- [x] Add webcam display functionality to `record` package, for use in custom CHS plugins (`video-config`, etc.) #40 
- [ ] Add webcam display extension to allow researchers to optionally display the webcam feed in non-CHS plugins

**Comment by mekline** [comment]
Update: waiting on generalized version for future sprint!

**Comment by becky-gilbert** [comment]
+1 for this from a researcher on slack 😄 (specifically for a config-quality trial)
